File: src/outlook-add-in/ReVision/src/taskpane/models/reviseResponse.ts
export class ReviseResponse {
  suggestionCategories: SuggestionCategory[];
  constructor(suggestionCategories: SuggestionCategory[]) {
    this.suggestionCategories = suggestionCategories;
  }
}

export class SuggestionCategory {
  category: string;
  categoryInUserLanguage: string;
  suggestions: Suggestion[];
  constructor(category: string, categoryInUserLanguage: string, suggestions: Suggestion[]) {
    this.category = category;
    this.categoryInUserLanguage = categoryInUserLanguage;
    this.suggestions = suggestions;
  }
}

export class Suggestion {
  needsAttention: boolean;
  title: string;
  titleInUserLanguage: string;
  explanation: string;
  explanationInUserLanguage: string;
  selected: boolean = false;
  constructor(
    needsAttention: boolean,
    title: string,
    titleInUserLanguage: string,
    explanation: string,
    explanationInUserLanguage: string
  ) {
    this.needsAttention = needsAttention;
    this.title = title;
    this.titleInUserLanguage = titleInUserLanguage;
    this.explanation = explanation;
    this.explanationInUserLanguage = explanationInUserLanguage;
  }
}

File: src/api/FunCoding.ReVision/FunCoding.ReVision.WebApi/Models/ReviseResponse.cs
namespace FunCoding.ReVision.WebApi.Models;

public class ReviseResponse
{
    public List<SuggestionCategory> SuggestionCategories { get; set; } = new();
}

public class SuggestionCategory
{
    public string Category { get; set; } = string.Empty;
    public List<Suggestion> Suggestions { get; set; } = new();
}

public class Suggestion
{
    public string Title { get; set; } = string.Empty;
    public string Explanation { get; set; } = string.Empty;
}
